### ISSUE TITLE
Code Standards

### ISSUE THREAD

**Issue by Greatexperiment** [deliciousmods/1956_reboot#11]
We need to agree on coding / comment / file standards that we can all agree on and will use. A big issue we have in 56 is that a lot of code is very lazily written, poorly formatted, badly organized, or just not commented enough. Let's fix that. 

This is a higher priority topic, since we need to establish this before starting the recode of any files. I will write up some standards of my own later, but I would like to see what everyone thinks first.

**Comment by Greatexperiment** [comment]
Let's start with comment standards:

1. Generally, comment often. Maybe this seems obvious, but it's extremely helpful to have a ton of comments in the code. Make it explicit, detailed, and easy for navigating in-file and via global searches.

2. Standard code comment style...

#####-----> File Header / Ender <><> <><> <><> <><> <><>

###---> Comment <><> <><> <><>

#-> Sub-Comment <><>

**Comment by Greatexperiment** [comment]
Examples:

#####-----> Air Techs File by Greatexperiment <><> <><> <><> <><> <><>

###---> Modded Jets <><> <><> <><>

#-> 1954 Jet <><>

#####-----> End File <><> <><> <><> <><> <><>

**Comment by eliasolsson** [comment]
Makes sense to me. I also think we should continue sorting stuff by country tags etc.

**Comment by Greatexperiment** [comment]
General structure:

r56_category_subcategory_name_name

It's long, but I think it's needed. It's way too easy to get lost in the code or files and not know what something is for or does. That's one reason we use the DXT5 graphics standard - it allows for quickly seeing what a file is without opening each one up in a graphics program. The new naming convention would make it very easy to do searches across the code for specific things, and would mean most anyone looking at it could easily figure out what something is associated with.
